helios-services/src/main/java/com/spotify/helios/agent/Supervisor.java

```diff
@@ -375,7 +385,13 @@ public void perform(final boolean done) throws InterruptedException {
           .setMaxIntervalMillis(SECONDS.toMillis(30))
           .build().newScheduler();
 
-      // Kill the container after stopping the runner
+
+      // TODO(negz): Use Docker stop instead of kill?
+      // I assume this loop is intended to handle the case where the above runner.stop() fails to
+      // communicate with the Docker API, because runner.stop() already sends a SIGKILL if
+      // necessary. What good could sending more SIGKILLs do if the first signal was sent
+      // successfully? If we *are* working around Docker API failures why not send more SIGTERMs
+      // to give the container the chance to shutdown gracefully?
```

Comment:
To be honest I am not sure about the intent of this loop here - maybe @rohansingh or @rculbertson might remember.